### PR TITLE
Document the binary protocol

### PR DESCRIPTION
**Pull request by elprans**
This adds the documentation of the wire format of the binary protocol.

### CODE REVIEW

File: docs/internals/protocol/dataformats.rst [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+        byte        data[length];
+    };
+
+
+.. _ref_protocol_fmt_uuid:
+
+std::uuid
+=========
+
+The :eql:type:`std::uuid` values are represented as an array of 16 unsigned
```
Comment (1st1) [inline docs/internals/protocol/dataformats.rst]:
```suggestion
The :eql:type:`std::uuid` values are represented as a sequence of 16 unsigned
```

File: docs/internals/protocol/dataformats.rst [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+
+
+.. _ref_protocol_fmt_datetime:
+
+std::datetime
+=============
+
+The :eql:type:`std::datetime` values are represented as a 64-bit integer,
+most sigificant byte first.  The value is the number of *microseconds*
+between the encoded datetime and January 1st 2000, 00:00 UTC.
```
Comment (1st1) [inline docs/internals/protocol/dataformats.rst]: Maybe it's worth mentioning that `946684800` is the *proper* number of seconds that should be used to convert between 2000/1970 timestamps.

File: docs/internals/protocol/messages.rst [hunk 3]
```diff
@@ -0,0 +1,926 @@
...
+        byte            data[32];
+    };
+
+
+.. _ref_protocol_msg_server_parameter_status:
+
+ParameterStatus
+===============
+
+Sent by: server
```
Comment (1st1) [inline docs/internals/protocol/messages.rst]: Nit: some "Sent by: server" end with a period and some don't.

**Comment by 1st1** [comment]
Should we add a "close" message? Or we do that in a separate pr?

**Comment by elprans** [comment]
> Should we add a "close" message? Or we do that in a separate pr?

In a separate PR that actually adds the message.

**Pull request closed without merging**